simulator/func_sim/func_sim.cpp

```diff
@@ -56,22 +56,39 @@ typename FuncSim<ISA>::FuncInstr FuncSim<ISA>::step()
 }
 
 template <typename ISA>
-Simulator::StopReason FuncSim<ISA>::run( uint64 instrs_to_run)
+std::pair<Simulator::StopReason, TrapType> FuncSim<ISA>::run( uint64 instrs_to_run)
```

Comment:
Would it be better to have a type alias for it?

Reply:
StopReason may be incorporated into TrapType directly, they are never used together.
Additionally, TrapType can be renamed to Trap, we know it is a type by its naming convention.